src/config_entries.c

```diff
@@ -108,7 +108,8 @@ static void config_entries_free(git_config_entries *entries)
 	list = entries->list;
 	while (list != NULL) {
 		next = list->next;
-		git__free((char *) list->entry->name);
```

Comment:
IMHO the usage of "key" in the commitmsg makes it less clear that name and key are the same thing.

```
Multivars are configuration entries that have many values for the same name;
we can thus micro-optimize this case by just retaining the name of the first
configuration entry and freeing all the others, letting them point to string of
the first entry.
```